packages/browser-utils/src/template-from-attributes.ts

```diff
@@ -0,0 +1,87 @@
+type AttributeMap = Record<string, string>;
+
+/**
+ * For each key in `attributes`, copies that attribute's value from `source`
+ * into all elements within `target` that match the corresponding value in
+ * `attributes`.
+ *
+ * For `<input type="checkbox">` elements it interprets the attribute as JSON
+ * and uses the truthiness of it to set `checked`. For other `<input>` elements,
+ * it sets the `value` attribute. For all others, it sets the `textContent`
+ * attribute.
+ *
+ * @param source The element to copy attributes from
+ * @param target The element to copy attributes into
+ * @param attributes A map of attributes to copy from `source` to `target`
+ * @param param.debug If true, logs debug information to the console
+ */
+export function templateFromAttributes(
+  source: HTMLElement,
+  target: HTMLElement,
+  attributes: AttributeMap,
+  { debug = false }: { debug?: boolean } = {},
+) {
+  function debugLog(...args: any[]): void {
+    if (debug) {
+      // eslint-disable-next-line no-console
+      console.log(...args);
+    }
+  }
+
+  Object.entries(attributes).forEach(([sourceAttribute, targetSelector]) => {
+    debugLog('---------------------------------------------------------------------');
+    debugLog('templateFromAttributes:');
+    debugLog('source', source);
+    debugLog('target', target);
+    debugLog('sourceAttribute', sourceAttribute);
+    debugLog('targetSelector', targetSelector);
```

Comment:
This is all copied verbatim from PrairieTest. It still hurts me a little bit to see all this, and IMO makes the function unnecessarily hard to read through, though I acknowledge that it's been useful to Matt in the past.

Reply:
I'd be happy with that! I'll add that.

Reply:
Changed in #e05b1d2.

Reply:
Nice use of `console.error()`!